internal/api/grpc_service.go

```diff
@@ -443,6 +442,62 @@ func (s *GrpcService) GetDeviceDefinitionAll(ctx context.Context, in *emptypb.Em
 	return result, nil
 }
 
+func (s *GrpcService) GetDeviceCompatibility(ctx context.Context, in *p_grpc.GetDeviceCompatibilityListRequest) (*p_grpc.GetDeviceCompatibilityListResponse, error) {
+	qryResult, _ := s.Mediator.Send(ctx, &queries.GetDeviceCompatibilityQuery{
+		MakeID:        in.MakeId,
+		IntegrationID: in.IntegrationId,
+		Region:        in.Region,
+	})
+
+	deviceCompatibilities := qryResult.(queries.GetDeviceCompatibilityQueryResult)
+
+	result := &p_grpc.GetDeviceCompatibilityListResponse{}
+
+	integFeats := deviceCompatibilities.IntegrationFeatures
+	dcMap := make(map[string][]*p_grpc.DeviceModelYears)
+	for _, v := range deviceCompatibilities.DeviceDefinitions {
+		// struct has noa values yet
+		if _, ok := dcMap[v.Model]; !ok {
+			dcMap[v.Model] = []*p_grpc.DeviceModelYears{}
+		}
+		if len(v.R.DeviceIntegrations) == 0 {
+			continue
+		}
+
+		di := v.R.DeviceIntegrations[0]
+
+		if di.Features.IsZero() {
+			continue
+		}
+		res := &p_grpc.DeviceModelYears{Year: int32(v.Year)}
+
+		feats := []*p_grpc.Feature{}
+		var dd []interface{}
+		err := di.Features.Unmarshal(&dd)
+		if err != nil {
+			continue
+		}
+
+		for _, i := range dd {
+			f := i.(map[string]interface{})
+			ft := &p_grpc.Feature{}
+			ft.Key = integFeats[f["feature_key"].(string)]
+			ft.SupportLevel = int32(f["support_level"].(float64))
```

Comment:
Actually, can we just re-use the struct we used to put these in the DB? That’s probably the safest thing.

I’m not sure this will work, as written with the snake case.